tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -338,24 +489,73 @@ void test_with_python(
   CAPTURE(dg_formulation);
   CAPTURE(use_moving_mesh);
   REQUIRE(face_mesh.number_of_grid_points() >= 1);
+  constexpr bool curved_background =
+      detail::has_inverse_spatial_metric_tag_v<System>;
   using dg_package_field_tags =
       typename BoundaryCorrection::dg_package_field_tags;
 
   MAKE_GENERATOR(gen);
   std::uniform_real_distribution<> dist(0.0, 1.0);
   DataVector used_for_size{face_mesh.number_of_grid_points()};
 
+  using face_tags = tmpl::list<FaceTags...>;
+  using face_tags_with_curved_background = tmpl::conditional_t<
+      curved_background,
+      tmpl::remove_duplicates<tmpl::push_back<
+          face_tags, typename TestHelpers::evolution::dg::detail::
+                         inverse_spatial_metric_tag<
+                             curved_background>::template f<System>>>,
+      face_tags>;
+
   Variables<dg_package_field_tags> package_data{used_for_size.size()};
-  const auto fields_on_face =
-      make_with_random_values<Variables<tmpl::list<FaceTags...>>>(
+  auto fields_on_face =
+      make_with_random_values<Variables<face_tags_with_curved_background>>(
           make_not_null(&gen), make_not_null(&dist), used_for_size);
   auto unit_normal_covector = make_with_random_values<
       tnsr::i<DataVector, FaceDim + 1, Frame::Inertial>>(
       make_not_null(&gen), make_not_null(&dist), used_for_size);
-  const auto normal_magnitude = magnitude(unit_normal_covector);
-  for (auto& component : unit_normal_covector) {
-    component /= get(normal_magnitude);
+  tnsr::I<DataVector, FaceDim + 1, Frame::Inertial> unit_normal_vector{};
+
+  if constexpr (not curved_background) {
+    const Scalar<DataVector> normal_magnitude = magnitude(unit_normal_covector);
+    for (auto& t : unit_normal_covector) {
+      t /= get(normal_magnitude);
+    }
+  } else {
+    using inv_spatial_metric = typename detail::inverse_spatial_metric_tag<
+        curved_background>::template f<System>;
+    // Make sure the inverse spatial metric is delta_{ij}+something smaller
+    // so that it's physically reasonable.
+    for (size_t i = 0; i < FaceDim + 1; ++i) {
```

Comment:
for this test to work, this same arbitrary inverse spatial metric choice needs to be identical in the two calls that are being compared, right? If so, it seems more maintainable to factor the assignment of the inverse spatial metric and the unit normal vector to another helper function so that if it is ever changed, it only needs to be changed in one place.